### PR TITLE
[WIP] python: Enforce line length limit (100)

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: python: Enforce line length limit (100)
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
not entirely clear why these are not picked up already and enforced by yapf

**Review by htuch — APPROVED** [review]
LGTM

File: .azure-pipelines/pipelines.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ stages:
             displayName: "Run format pre-checks"
 
       - job: format
-        dependsOn: ["format_pre"]
+        dependsOn: [] # "format_pre"]
```
Comment (phlax) [inline .azure-pipelines/pipelines.yml:31]: remove

**Comment by phlax** [comment]
im still of the opinion that we should only run the autoformatter if flake8 fails (and only on those lines)

i think this demonstrates quite well why

```diff
-        return (
-            (file_path.endswith('.h')
-             and not file_path.startswith("./test/")
-             and not file_path in EXCEPTION_ALLOWLIST)
-            or file_path in EXCEPTION_DENYLIST
-            or self.is_in_subdir(file_path, 'tools/testdata'))
+        return ((
+            file_path.endswith('.h') and not file_path.startswith("./test/")
+            and not file_path in EXCEPTION_ALLOWLIST) or file_path in EXCEPTION_DENYLIST
+                or self.is_in_subdir(file_path, 'tools/testdata'))


```

**Comment by phlax** [comment]
also this is still going to require a bit more work as i didnt fix most of the long lines yet (i opened #15587 to make it easier to do so)

**Pull request closed without merging**